scripts/us_cdc/natality/preprocess.py

```diff
@@ -0,0 +1,221 @@
+                val = row[col]
+                if val in _CONFIG['pvs'][col]:  # If pv is defined
+                    statvar.update(_CONFIG['pvs'][col][val])
+                else:
+                    skip_statvar = True
+                    break  # Skip this row
+
+            if not skip_statvar:
+                statvar['Node'] = get_statvar_dcid(statvar)
+                processed_dict = {
+                    'Year': year,
+                    'Geo': geo_dcid,
+                    'StatVar': statvar['Node'],
+                    'Quantity': row[data_col],
+                    'Unit': statvar.get('unit', None)
+                }
+                csv_writer.writerow(processed_dict)
+                statvars.append(statvar)
+
+    return statvars
+
+
+def is_quantity_range(val: str) -> bool:
+    """Checks if [] are present in val. While this is not a perfect check for
+    quantity ranges, it works in this case."""
+    if '[' in val and ']' in val:
+        return True
+    return False
+
+
+def write_to_mcf(sv_list: list, mcf_path: str):
+    """Writes all statvars to a .mcf file."""
+    if sv_list:
+        dcid_set = set()
+        with open(mcf_path, 'w', encoding='utf-8') as f:
+            for sv in sv_list:
+                dcid = sv['Node']
+                if dcid in dcid_set:
+                    continue
+                dcid_set.add(dcid)
+                statvar_mcf_list = [f'Node: dcid:{dcid}']
+                for p, v in sv.items():
+                    if p not in ['Node', 'unit']:
+                        if is_quantity_range(v) or p == 'description':
+                            statvar_mcf_list.append(f'{p}: {v}')
+                        else:
+                            statvar_mcf_list.append(f'{p}: dcs:{v}')
+                statvar_mcf = '\n'.join(statvar_mcf_list) + '\n\n'
+                f.write(statvar_mcf)
+
+
+def main(argv):
+    global _CONFIG, _SKIP_YEARS
+    with open(_FLAGS.config_path, 'r', encoding='utf-8') as config_f:
+        _CONFIG = json.load(config_f)
+
+    if _FLAGS.skip_years:
+        _SKIP_YEARS = _FLAGS.skip_years
+
+    cleaned_csv_path = os.path.join(_FLAGS.output_path, 'cleaned.csv')
+    output_mcf_path = os.path.join(_FLAGS.output_path, 'output.mcf')
+
+    with open(cleaned_csv_path, 'w', encoding='utf-8') as cleaned_f:
+
+        f_writer = csv.DictWriter(cleaned_f, fieldnames=_OUTPUT_COLUMNS)
+        f_writer.writeheader()
+
+        statvar_list = []
+        for file_name in os.listdir(_FLAGS.input_path):
+            if file_name.endswith('.txt'):  # Raw data is .txt files
+                file_path = os.path.join(_FLAGS.input_path, file_name)
+                with open(file_path, 'r', encoding='utf-8') as csv_f:
+                    f_reader = csv.DictReader(csv_f,
+                                              delimiter=',',
+                                              quotechar='"')
+                    f_name = file_name[:-4]  # Removing .txt from filename
```

Comment:
Can use os.path.splitext() to remove extension

Reply:
Done.